src/ess/sans/normalization.py

```diff
@@ -212,22 +213,31 @@ def iofq_norm_wavelength_term(
         Used by :py:func:`iofq_denominator`.
     """
     out = incident_monitor * transmission_fraction
-    if direct_beam is not None:
-        # Make wavelength the inner dim
-        dims = list(direct_beam.dims)
-        dims.remove('wavelength')
-        dims.append('wavelength')
-        direct_beam = direct_beam.transpose(dims)
-        out = direct_beam * broadcast_uncertainties(
-            out, prototype=direct_beam, mode=uncertainties
-        )
     # Convert wavelength coordinate to midpoints for future histogramming
     out.coords['wavelength'] = sc.midpoints(out.coords['wavelength'])
-    return NormWavelengthTerm[ScatteringRunType](out)
+    return MonitorTerm[ScatteringRunType](out)
+
+
+def norm_detector_term(
+    solid_angle: MaskedSolidAngle[ScatteringRunType],
+    direct_beam: CleanDirectBeam,
+    uncertainties: UncertaintyBroadcastMode,
+) -> CleanWavelength[ScatteringRunType, Denominator]:
+    # Make wavelength the inner dim
+    dims = list(direct_beam.dims)
+    dims.remove('wavelength')
+    dims.append('wavelength')
+    direct_beam = direct_beam.transpose(dims)
+    out = solid_angle * broadcast_uncertainties(
+        direct_beam, prototype=solid_angle, mode=uncertainties
+    )
+    # Convert wavelength coordinate to midpoints for future histogramming
+    out.coords['wavelength'] = sc.midpoints(out.coords['wavelength'])
+    return CleanWavelength[ScatteringRunType, Denominator](out)
 
 
 def iofq_denominator(
```

Comment:
TODO: This function is unused now, remove it (but keep math in docstring somewhere).